opm/simulators/wells/MultisegmentWellGeneric.cpp

```diff
@@ -63,8 +65,15 @@ MultisegmentWellGeneric(WellInterfaceGeneric& baseif)
     baseif.perfDepth().resize(baseif_.numPerfs(), 0.);
     for (size_t perf = 0; perf < completion_set.size(); ++perf) {
         const Connection& connection = completion_set.get(perf);
-        if (connection.state() == Connection::State::OPEN) {
+        if (connection.state() == Connection::State::OPEN && connection.attachedToSegment()) {
             const int segment_index = segmentNumberToIndex(connection.segment());
+            if ( segment_index == -1) {
+                OPM_THROW(std::logic_error,
+                          fmt::format("COMPSEGS: Well {} has connection in cell {}, {}, {} "
+                                      "without associated segment.", baseif_.wellEcl().name(),
+                                      connection.getI(), connection.getJ(),
+                                      connection.getK()));
```

Comment:
Tiny nit: Could you output `getI() + 1` (etc.) instead, please?  The `Connection::getX()` functions return zero-based indices, and users are more accustomed to one-based indices in this context.

Reply:
Don't even consider this a nit. Fixed.